### PR TITLE
[WIP] automation

### PR DESCRIPTION
**Pull request by jualiasha**

### CODE REVIEW

File: dashboard/public/automation.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+	"automation": [
+		{
+			"category": "Internet",
+			"topic": "How to connect to Telia",
+			"description": "You need to do several things, Lorem ipsum",
+			"id": 1
```
Comment (Romkin7) [inline dashboard/public/automation.json:8]: add here categories list as array. in all automation items. we can build response from backend like this json and send it

**Review by Romkin7 — CHANGES_REQUESTED** [review]

File: dashboard/src/Pages/Automation/Automation.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	useEffect(() => {
+		if (!automation) {
+			axios.get('automation.json').then((res: any) => {
+				console.log(res.data.automation);
+				setAutomation(res.data.automation);
+			});
+		}
+	}, [setAutomation, automation]);
+
+	const allcategories = automation?.map((automation: any) => automation.category);
```
Comment (Romkin7) [inline dashboard/src/Pages/Automation/Automation.tsx]: we can't have array name in singular form it should be auomations.map((automation)=> {})

so array name is automations in plural form.

**Review by Romkin7 — CHANGES_REQUESTED** [review]

File: dashboard/public/automation.json (hunk 1 above)
Comment (jualiasha) [reply dashboard/public/automation.json:8]: done

File: dashboard/src/Pages/Automation/Automation.tsx (hunk 2 above)
Comment (jualiasha) [reply dashboard/src/Pages/Automation/Automation.tsx]: done

File: dashboard/src/Pages/Automation/Automation.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import React, { FC, useState, useEffect } from 'react';
+import { IAutomation } from '../../@types';
+import Categories from '../../Components/Main/Automation/Categories';
+import Topics from '../../Components/Main/Automation/Topics';
+import axios from 'axios';
+import { AddCircleOutline } from '@material-ui/icons';
+import { Route, Switch, useLocation, useParams } from 'react-router-dom';
```
Comment (Romkin7) [inline dashboard/src/Pages/Automation/Automation.tsx]: Remove Route, Switch, useLocation, useParams from here. we obly sahould have Switch in Main.tsx and everywhere else only Link components from react-router-dom

**Review by Romkin7 — CHANGES_REQUESTED** [review]

File: dashboard/src/Pages/Automation/Automation.tsx (hunk 3 above)
Comment (Romkin7) [reply dashboard/src/Pages/Automation/Automation.tsx]: But use here for this conditional rendering, routes are not needed here.

File: dashboard/src/Pages/Automation/Automation.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+	const faqcategorylist: any = automations?.filter((automation) => {
+		return automation.category.includes('FAQ');
+	});
+
+	console.log(faqcategorylist);
+
+	const handleCategoryClick = (category: string) => {
+		console.log('clicked ' + category);
+		const filteredcategorylist: any = automations?.filter((automation) => {
```
Comment (Romkin7) [inline dashboard/src/Pages/Automation/Automation.tsx]: /** Filter returns whole object based on condition, while map can return only some parts of object like only categories of automation object and leave out all the rest.
	 * We can also chain map and filter, like i do here, i do map on filtered array of automation objects and return only categories.
	 */

**Review by Romkin7 — CHANGES_REQUESTED** [review]

**Review by Romkin7 — APPROVED** [review]

File: dashboard/src/Pages/Automation/Automation.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+					<AddCircleOutline style={{ fontSize: 30 }} className="automation__category-list__button__icon" role="button" />
+				</h2>
+				{categories?.length &&
+					categories.map((category: string) => {
+						return (
+							<Categories
+								key={category}
+								category={category}
+								click={() => handleCategoryClick(category)}
+								active={
```
Comment (Romkin7) [inline dashboard/src/Pages/Automation/Automation.tsx]: active should be boolean prop not class. active true or false. based on that boolean value add class active or not active.

**Review by Romkin7 — CHANGES_REQUESTED** [review]

File: dashboard/src/Components/Topic/Topic.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { FiberManualRecord } from '@material-ui/icons';
+import React, { FC } from 'react';
+import { Link, Route, Switch, useParams } from 'react-router-dom';
+
+const Topics: FC<any> = ({ topic }) => {
```
Comment (Romkin7) [inline dashboard/src/Components/Topic/Topic.tsx]: Please don't use any in for props types, try avoid any type unless you can't do anything else and can't move on without it.

Comment (Romkin7) [reply dashboard/src/Components/Topic/Topic.tsx]: no any types for props, create interfaces instead as i did for Category.tsx
```
interface ICategoryProps {
	category: string;
	categoryClickHandler: (category: string) => void;
	active: boolean | null;
}

const Category: FC<ICategoryProps> = ({ category, categoryClickHandler, active }) => {
	return (
```

File: dashboard/src/Components/Topic/Topic.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { FiberManualRecord } from '@material-ui/icons';
+import React, { FC } from 'react';
+import { Link, Route, Switch, useParams } from 'react-router-dom';
```
Comment (Romkin7) [inline dashboard/src/Components/Topic/Topic.tsx]: remove Route and Switch from here, we don't use Route or Switch outside of Main.tsx Switch and more Switch than that one.

**Review by Romkin7 — CHANGES_REQUESTED** [review]

**Review by Romkin7 — APPROVED** [review]

**Pull request merged**